Python/tier2_redundancy_eliminator_bytecodes.c

```diff
@@ -220,7 +220,7 @@ dummy_func(void) {
     }
 
     op(_TO_BOOL_NONE, (value -- res)) {
-        if (sym_get_const(value) == Py_None) {
+        if (sym_is_const(value) && (sym_get_const(value) == Py_None)) {
```

Comment:
Why is this necessary? Shouldn't `sym_get_const` return `NULL` for anything that isn't a constant?
It does seem rather verbose to have to prefix every `sys_get_const` with a `sys_is_const`.

Reply:
In the code:
```C
static inline PyObject *
sym_get_const(_Py_UOpsSymType *sym)
{
    assert(sym_is_const(sym));
    assert(sym->const_val);
    return sym->const_val;
}
```
So we assume there is always something there when there's `sym_get_const`. I think I can just change the contract to return NULL on non constants though.

Reply:
The new sym_get_const() indeed returns NULL in this case.